### PR TITLE
Update developer documentation

### PR DESCRIPTION
**Pull request by ximinez**
I tried to update, add, and correct all the docs in apply.h and applySteps.h. In particular, `ripple::apply` gives an overview of the steps of the process.

### CODE REVIEW

File: src/ripple/app/tx/applySteps.h [hunk 1]
```diff
@@ -82,32 +106,43 @@ struct PreclaimResult
...
         std::pair<TER, std::uint64_t> const& result)
         : PreclaimResult(ctx_, result.first, result.second)
     {
     }
 
+    /// Deleted copy assignment operator
     PreclaimResult& operator=(PreclaimResult const&) = delete;
 };
 
+/** Structure describing the account consequences if
```
Comment (bachase) [inline src/ripple/app/tx/applySteps.h]: How should I understand  `account consequences` if the class is called `TxConsequences`?

**Comment by ximinez** [comment]
Ironically, given the nature of this PR, I forgot the include the rendered documentation link:

https://ximinez.github.io/apply_doc/

File: src/ripple/app/tx/apply.h [hunk 2]
```diff
@@ -56,38 +65,57 @@ checkValidity(HashRouter& router,
 
 
 /** Sets the validity of a given transaction in the cache.
-    Use with extreme care.
+
+    @warn Use with extreme care.
```
Comment (ximinez) [inline src/ripple/app/tx/apply.h]: Oops. The correct tag is `@warning`. I don't see it rendering much differently, but oh well.

File: src/ripple/app/tx/apply.h [hunk 3]
```diff
@@ -56,38 +65,57 @@ checkValidity(HashRouter& router,
...
+/** Apply a transaction to an OpenView.
+
+    This function is the canonical way to apply a transaction
+    to a ledger. It rolls the validation and application
+    steps into one function. To do the steps manually, the
+    correct calling order is:
+    @code{.cpp}
+    preflight -> preclaim -> doApply
+    @endcode
+    And the result of one function must be passed to the next.
```
Comment (ximinez) [inline src/ripple/app/tx/apply.h]: Will do.

File: src/ripple/app/tx/apply.h [hunk 4]
```diff
@@ -56,38 +65,57 @@ checkValidity(HashRouter& router,
...
+    The `preflight` result can be safely cached and reused
+    asynchronously, but `preclaim` and `doApply` must be called
+    in the same thread and with the same view.
+
+    @throws none Does not throw.
```
Comment (ximinez) [inline src/ripple/app/tx/apply.h]: Yeah, I wasn't super happy with that either. Maybe I'll just do it as a `@note`...

File: src/ripple/app/tx/applySteps.h (hunk 1 above)
Comment (ximinez) [inline src/ripple/app/tx/applySteps.h]: Good question. I rephrased.

**Comment by ximinez** [comment]
Squashed and "Passed".

**Comment by nbougalis** [comment]
Merged as fcceb0aac10c1cbd12c0dc3e3a2d21b711e64497.

**Pull request closed without merging**